src/GenSequence.ts

```diff
@@ -27,6 +27,7 @@ export interface Sequence<T> extends IterableLike<T> {
     any(fnFilter: (t: T)=> boolean): boolean;
     first(fnFilter?: (t: T)=> boolean, defaultValue?: T): Maybe<T>;
     first(fnFilter: (t: T)=> boolean, defaultValue: T): T;
+    max(): Maybe<T>;
```

Comment:
I started to rearrange things based on this order, but after looking at it, I will send out a separate pull request that is just re-ordering so it is easier to review.